Adding output to syslog

bolemo opened the pull request:
Slight changes:
– to have the script output to log in syslog, as well as
– print the output to stdout (as before) and stderr (when it is an error message)

bolemo committed:
Adding output to syslog

frizkyiman reviewed (changes requested):
Hi there! thanks for help me improve this script.
it would be good to keep old echo too beside the output to syslog for inform user when they want to run the script manually, so the err or output will be print to terminal as well.
its good or is it a bad idea to include both?

bolemo commented:
Hi there :)

In my change, it echos to sdtout/stderr and syslog (both) :)

bolemo reviewed (commented):
The function  `_echo` already does both:
Output echo to stdout/stderr **and** to syslog :)

frizkyiman commented:
> The function `_echo` already does both: Output echo to stdout/stderr **and** to syslog :)

i mean the message show to like reguler echo when run that script via ssh like this, so the script has 2 echo to stdout/stderr on syslog and show message to user if they run the script manually via ssh.

![photo_2024-03-23_09-56-46](https://github.com/frizkyiman/fix-read-only/assets/110967316/27be6c7d-2fb2-4a2c-a3f2-ad10a5b29a7c)
![photo_2024-03-23_10-00-23](https://github.com/frizkyiman/fix-read-only/assets/110967316/d21525c5-3262-4f8d-90f3-c9c1bcc69ca4)

frizkyiman commented:
i think this good way to inform both syslog and user to easy for debug ?
please let me know about your opinion to this. 

> 
_echo err "Root file system device '$root_device' not found in the system. Unable to repair."
echo "Root file system device '$root_device' not found in the system. Unable to repair."

bolemo committed:
Update repair_ro

Fix bug where instead of echoing to actual stderr or stdout, it was echoing to files named '&1' or '&2'

bolemo committed:
Update repair_ro

Simplified _echo function

bolemo commented:
@frizkyiman 

I understand what happened!

My `_echo` function was supposed to do both already (echo and syslog), but I just tested, and I had no output to stdout/err, so I understand better your questions.

I found the reason: it was writing the output to files named `'&1'`, `'&2'` instead of the actual output descriptors (variable $O was treated as a file path, not literally). I simplified `_echo` it and tested it, and it works as supposed now:
```
echo() {
  [ "$1" = 'err' ] && echo $2 >&2 || echo $2
  logger -p "$1" -t repair_co "$2"
}
```
And the test:
```
root@Zeus:~# repair_ro 
Root file system device '/dev/nvme0n1p2' is currently in read-write mode.

root@Zeus:~# repair_ro /dev/notthere
Root file system device '/dev/notthere' not found in the system. Unable to repair.

root@Zeus:~# logread | grep -F repair
Sat Mar 23 11:40:58 2024 user.notice repair_co: Root file system device '/dev/nvme0n1p2' is currently in read-write mode.
Sat Mar 23 11:40:59 2024 user.err repair_co: Root file system device '/dev/notthere' not found in the system. Unable to repair.
```

I am sorry, I should have tested more deeply before my initial commit.
Now it is working as was supposed to be : it outputs the messages to stdout/stderr **and** syslog
And there is **no need** to add an extra echo like this:
```
_echo err "Root file system device '$root_device' not found in the system. Unable to repair."
echo "Root file system device '$root_device' not found in the system. Unable to repair."
```
As the whole idea of the `_echo` function was from the beginning to do _two birds one stone_ and avoid repetition in the code ;)

frizkyiman commented:
Great!
thanks for fix and test :)

will merged ASAP!
